File: src/main/kotlin/dhbw/karlsruhe/gitgood/database/DatabaseConnection.java
package dhbw.karlsruhe.gitgood.database;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.sql.Statement;

import lombok.extern.slf4j.Slf4j;

@Slf4j
public class DatabaseConnection {

    private static Connection connection;
    private static Statement statement;

    public static void initConnection(String connectionUrl, String userName,  String password) {
        try {
            connection = DriverManager.getConnection(connectionUrl, userName, password);
            statement = connection.createStatement();
            log.info("Database connection sucessfull!");
        } catch (SQLException e){
            log.warn(e.getMessage());
        }
    }

    public static Connection getConnection() {
        return connection;
    }

    public static Statement getStatement() {
        return statement;
    }
}

File: src/main/kotlin/dhbw/karlsruhe/gitgood/database/DatabaseConnection.kt
package dhbw.karlsruhe.gitgood.database

import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import java.sql.Connection
import java.sql.DriverManager
import java.util.*

object DatabaseConnection {

    lateinit var connection : Connection

    fun initConnection(connectionUrl: String, username: String, pw: String) {

        val properties = Properties()

        // Populate the properties file with user name and password
        with(properties) {
            put("user", username)
            put("password", pw)
        }

        // Open a connection to the database
        connection = DriverManager.getConnection(connectionUrl, properties)
        println("Database connection sucessfull!")
    }

}